bundles/core/org.eclipse.smarthome.core.test/src/test/java/org/eclipse/smarthome/core/service/StateDescriptionServiceImplTest.java

```diff
@@ -0,0 +1,265 @@
+	private static final BigDecimal STATE_DESCRIPTION_PROVIDER_DEFAULT_STEP = new BigDecimal("0");
+	private static final String STATE_DESCRIPTION_PROVIDER_DEFAULT_PATTERN = "pattern1";
+	private static final boolean STATE_DESCRIPTION_PROVIDER_DEFAULT_IS_READONLY = false;
+	private static final List<StateOption> STATE_DESCRIPTION_PROVIDER_DEFAULT_OPTIONS = Collections.emptyList();
+
+	private StateDescriptionServiceImpl mergingService = new StateDescriptionServiceImpl();
+	private NumberItem item;
+
+	@Test
+	public void testServiceWithOneStateDescriptionProvider() {
+		StateDescriptionProvider stateDescriptionProviderDefault = mock(StateDescriptionProvider.class);
+		when(stateDescriptionProviderDefault.getRank()).thenReturn(STATE_DESCRIPTION_PROVIDER_DEFAULT_SERVICE_RANKING);
+		StateDescription stateDescription1 = new StateDescription(STATE_DESCRIPTION_PROVIDER_DEFAULT_MIN_VALUE,
+				STATE_DESCRIPTION_PROVIDER_DEFAULT_MAX_VALUE, STATE_DESCRIPTION_PROVIDER_DEFAULT_STEP,
+				STATE_DESCRIPTION_PROVIDER_DEFAULT_PATTERN, STATE_DESCRIPTION_PROVIDER_DEFAULT_IS_READONLY,
+				STATE_DESCRIPTION_PROVIDER_DEFAULT_OPTIONS);
+		when(stateDescriptionProviderDefault.getStateDescription(ITEM_NAME, null)).thenReturn(stateDescription1);
+
+		mergingService.addStateDescriptionProvider(stateDescriptionProviderDefault);
+
+		item = new NumberItem(ITEM_NAME);
+		item.setStateDescriptionService(mergingService);
+		StateDescription finalStateDescription = item.getStateDescription();
+
+		assertThat(finalStateDescription.getMinimum(), is(STATE_DESCRIPTION_PROVIDER_DEFAULT_MIN_VALUE));
+		assertThat(finalStateDescription.getMaximum(), is(STATE_DESCRIPTION_PROVIDER_DEFAULT_MAX_VALUE));
+		assertThat(finalStateDescription.getStep(), is(STATE_DESCRIPTION_PROVIDER_DEFAULT_STEP));
+		assertThat(finalStateDescription.getPattern(), is(STATE_DESCRIPTION_PROVIDER_DEFAULT_PATTERN));
+		assertThat(finalStateDescription.isReadOnly(), is(STATE_DESCRIPTION_PROVIDER_DEFAULT_IS_READONLY));
+		assertThat(finalStateDescription.getOptions(), is(STATE_DESCRIPTION_PROVIDER_DEFAULT_OPTIONS));
+	}
+
+	@Test
+	public void testMinValueMaxValueStepAndPatternTwoDescriptionProviders() {
+		int stateDescriptionProvider1ServiceRanking = -1;
+		int stateDescriptionProvider2ServiceRanking = -2;
+
+		BigDecimal stateDescriptionProvider1MinValue = new BigDecimal("-1");
+		BigDecimal stateDescriptionProvider2MinValue = new BigDecimal("-2");
+
+		BigDecimal stateDescriptionProvider1MaxValue = new BigDecimal("-1");
+		BigDecimal stateDescriptionProvider2MaxValue = new BigDecimal("-2");
+
+		BigDecimal stateDescriptionProvider1Step = new BigDecimal("-1");
+		BigDecimal stateDescriptionProvider2Step = new BigDecimal("-2");
+
+		String stateDescriptionProvider1Pattern = "pattern1";
+		String stateDescriptionProvider2Pattern = "pattern2";
+
+		StateDescriptionProvider stateDescriptionProvider1 = mock(StateDescriptionProvider.class);
```

Comment:
Since the following lines down to `StateDescription finalStateDescription = item.getStateDescription();` are always the same for each test, how about creating a function that you pass the rank and description form 2 providers and return the `finalStateDescription`, to avoid the duplication of the code?